### DISCUSSION TITLE
qpsol computation time and options

### DISCUSSION THREAD

**Discussion by SAA1995** [casadi/casadi#3301] [Q&A]
Hi everyone,

Im new to Casadi and planning on using it for my PhD thesis. 

I have been working on solving a QP problem for one of my controllers and I would like to see the computation time it requires for solving the problem. i want to store this time just like the QP problem output as a variable. And I believe majority of the stats for the solver are available in solver.stats() but I cant seem to get it to work. 
Furthermore, I would also like to specify options to the qpsol() of like max iterations, print options etc. but it appears to me that they are different from the options struct given to nlpsol(). Attached below is my attempt to make options work with qpsol, but it doesnt work.

```
prob = struct('f', J, 'x', vertcat(w{:}), 'g', vertcat(g{:}),'p',P);
options = struct('qpoases',struct('print_level',0),'print_time',false); % trying to switch the print time off since im using the generated code in simulink and it slows computation

solver = qpsol('solver', 'qpoases', prob,options); % Doesnt work!
```

I am using the MATLAB interface for Casadi, with the traditional MX method of making problem and then using code generator to convert the casadi code to c code for simulink s function ([using this blog post](https://web.casadi.org/blog/mpc-simulink2/)). I would be really grateful if someone can point me to some old thread or an answer for the syntax for specifying options to qpsol in MATLAB. And also, how to save the computation time from solver.stats to a vector. 

Thanks for your time and help.

**Comment by jgillis** [comment]
Options like print level, max iter, etc are solver specific.
When we can, our interfaces just pass options as-is to the solver without knowledge of their names or meaning.
Sometimes, the solver does not allow this and we do need to enumerate the possibilities in our interface.
Qpoases is of the latter type, meaning that the options get documented: https://web.casadi.org/python-api/#qpoases
You need `'none'` for print level.

You may need `'record_time', true` to get timing stats
